mobile/ios/ios-core.ts

```diff
@@ -980,30 +983,80 @@ export class PlistService implements Mobile.IiOSDeviceSocket {
 	}
 }
 
-export class GDBServer implements Mobile.IGDBServer {
-	private socket: Mobile.IiOSDeviceSocket  = null;
+util.inherits(GDBStandardOutputAdapter, stream.Transform);
+function GDBStandardOutputAdapter(opts?: stream.TransformOptions) {
+	stream.Transform.call(this, opts);
+	this.utf8StringDecoder = new string_decoder.StringDecoder("utf8");
+}
+GDBStandardOutputAdapter.prototype._transform = function(packet: Buffer, encoding: string, done: Function): void {
+	var result = "";
 
-	constructor(private service: number,
-		private $injector: IInjector) {
+	for (var i = 0; i < packet.length; i++) {
+		if (packet[i] === "$".charCodeAt(0)) {
+			var start = ++i;
 
-		if(hostInfo.isWindows()) {
-			this.socket = this.$injector.resolve(WinSocket, {service: this.service, format: 0});
-		} else if(hostInfo.isDarwin()) {
-			this.socket = this.$injector.resolve(PosixSocket, {service: this.service, format: CoreTypes.kCFPropertyListXMLFormat_v1_0});
+			while (packet[i] !== "#".charCodeAt(0)) {
```

Comment:
It should be accessable from `GDBStandardOutputAdapter` and `GDBSignalWatcher` so I suggest to make it internal for `ios-core` file